Add Smithery CLI Installation Instructions

calclavia opened the pull request:
This PR adds installation instructions to automatically install BigQuery MCP Server for Claude Desktop using Smithery CLI. This makes it easier for users to install the MCP. Also, it adds a badge to show the number of installations from Smithery.

ergut commented:
Thanks for adding Smithery integration to make the installation process smoother! I noticed a few issues when testing it out.

Currently, when ran as follows:
```
% npx @smithery/cli install @ergut/mcp-bigquery-server --client claude
Need to install the following packages:
@smithery/cli@0.0.12
Ok to proceed? (y) 

? Please enter value for projectId: my-project
? Please enter value for location: us-central1
Updated claude configuration
? Would you like to help improve @smithery/cli by sharing anonymous installation analytics? Yes
? Would you like to restart the claude app to apply changes? Yes
Restarting claude app...
Claude has been restarted.
```
 the Smithery CLI generates this configuration:
```javascript
"@ergut/mcp-bigquery-server": {
    "command": "npx",
    "args": [
        "-y",
        "@ergut/mcp-bigquery-server",
        "my-project",
        "us-central1"
    ],
    "env": {
        "projectId": "my-project",
        "location": "us-central1"
    }
}
```

There are two things we need to address:

1. The `args` format needs updating. In v1.0.0 we switched from positional to named arguments:
```javascript
"args": [
    "-y",
    "@ergut/mcp-bigquery-server",
    "--project-id",
    "my-project",
    "--location",
    "us-central1"
]
```

2. The `env` block is not needed and should be removed.

Additionally, it would be helpful to add support for the optional `--key-file` parameter in the Smithery CLI prompts. We introduced this option to support service account authentication, which many organizations prefer over user credentials.

Let me know if you need any clarification on these changes!

calclavia commented:
> Thanks for adding Smithery integration to make the installation process smoother! I noticed a few issues when testing it out.
> 
> Currently, when ran as follows:
> 
> ```
> % npx @smithery/cli install @ergut/mcp-bigquery-server --client claude
> Need to install the following packages:
> @smithery/cli@0.0.12
> Ok to proceed? (y) 
> 
> ? Please enter value for projectId: my-project
> ? Please enter value for location: us-central1
> Updated claude configuration
> ? Would you like to help improve @smithery/cli by sharing anonymous installation analytics? Yes
> ? Would you like to restart the claude app to apply changes? Yes
> Restarting claude app...
> Claude has been restarted.
> ```
> 
> the Smithery CLI generates this configuration:
> 
> ```js
> "@ergut/mcp-bigquery-server": {
>     "command": "npx",
>     "args": [
>         "-y",
>         "@ergut/mcp-bigquery-server",
>         "my-project",
>         "us-central1"
>     ],
>     "env": {
>         "projectId": "my-project",
>         "location": "us-central1"
>     }
> }
> ```
> 
> There are two things we need to address:
> 
> 1. The `args` format needs updating. In v1.0.0 we switched from positional to named arguments:
> 
> ```js
> "args": [
>     "-y",
>     "@ergut/mcp-bigquery-server",
>     "--project-id",
>     "my-project",
>     "--location",
>     "us-central1"
> ]
> ```
> 
> 2. The `env` block is not needed and should be removed.
> 
> Additionally, it would be helpful to add support for the optional `--key-file` parameter in the Smithery CLI prompts. We introduced this option to support service account authentication, which many organizations prefer over user credentials.
> 
> Let me know if you need any clarification on these changes!

Thanks for spotting the issue! I fixed the config example and prevented it from adding empty ENV vars, you can take a look here:
https://smithery.ai/protocol/@ergut/mcp-bigquery-server

ergut commented:
Thanks for fixing the configuration format! I'll merge this PR as it resolves the main issues. For future reference, you might want to update the transport configuration to
- Put the required project id parameter first (currently it is the second parameter)
- Include the key-file option for users who need service account authentication

Thanks for your contribution!
